source/state_representation/include/state_representation/space/Jacobian.hpp

```diff
@@ -222,27 +236,36 @@ class Jacobian : public State {
    * @brief Return the pseudoinverse of the Jacobian matrix
    * @return The pseudoinverse of the Jacobian
    */
-  Jacobian pseudoinverse() const;
+  Eigen::MatrixXd pseudoinverse() const;
 
   /**
-   * @brief Solve the system X = inv(J)*M to obtain X which is more efficient than multiplying with the pseudo-inverse
-   * @param matrix The matrix to solve the system with
-   * @return The result of X = J.solve(M) from Eigen decomposition
+   * @brief Multiply the given matrix by the pseudoinverse of the Jacobian matrix
+   * @param matrix The matrix to be multiplied with the Jacobian
+   * @return The result of the matrix multiplication
    */
-  Eigen::MatrixXd solve(const Eigen::MatrixXd& matrix) const;
+  Eigen::MatrixXd pseudoinverse(const Eigen::MatrixXd& matrix) const;
 
   /**
-   * @brief Solve the system dX = J*dq to obtain dq which is more efficient than multiplying with the pseudo-inverse
-   * @param dX The Cartesian twist to multiply with
+   * @brief Transform the given Cartesian twist to joint space by multiplication with the pseudoinverse of the
+   * Jacobian matrix
```

Comment:
nitpick: I don't like "the given" in docstrings. It's over-specifying something generic. Also, the rest of the sentence can be made more compact and accurate as follows:
```suggestion
   * @brief Transform a Cartesian twist to joint space by pre-multiplying the Jacobian pseudoinverse
```
Same suggestion to apply to CartesianWrench transform.